### PR TITLE
Adds regression test for cloudNative with codeModules and emulated network problems

### PR DESCRIPTION
**Pull request by waodim**
# Description
This test should ensure that the csi driver works as intended. Due to network problems the driver should attach dummy volumes to the sample pods, in order for them to still run and not crash or be stuck.

## How can this be tested?
First install the cilium cli, run `cilium install` after creating a cluster and then simply run the tests.

## Checklist
- [x] PR is labeled accordingly
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: test/cloudnative/network_problems_test.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+		require.NoError(t, err)
+		assert.Contains(t, result.StdOut.String(), errorMessage)
+	}
+	return ctx
+}
+
+func checkPodsAfterSleep(ctx context.Context, t *testing.T, environmentConfig *envconf.Config) context.Context {
+	resources := environmentConfig.Client().Resources()
+
+	samplePods := sampleapps.Get(t, ctx, resources)
```
Comment (meik99) [inline test/cloudnative/network_problems_test.go:98]:
```suggestion
	resources := environmentConfig.Client().Resources()
	samplePods := sampleapps.Get(t, ctx, resources)
```

File: test/cloudnative/network_problems_test.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+			WithDefaultObjectMeta().
+			FeatureFlag(featureFlag).
+			ApiUrl(secretConfigs[0].ApiUrl).
+			CloudNative(codeModulesSpec()).
+			Build()),
+	)
+
+	createNetworkProblems.Assess("install deployment", manifests.InstallFromFile("../testdata/cloudnative/codemodules-deployment.yaml"))
+
+	createNetworkProblems.Assess("start sample apps and injection", sampleapps.Install)
```
Comment (meik99) [inline test/cloudnative/network_problems_test.go:66]: afaik functions supplied with `.Setup` are always called before functions supplied with `.Assess`. Therefore they should be ordered as such in code
```suggestion
	createNetworkProblems.Setup(secrets.ApplyDefault(secretConfigs[0]))
	createNetworkProblems.Setup(operator.InstallAllForKubernetes())
	
	setup.AssessDeployment(createNetworkProblems)

	createNetworkProblems.Assess("apply network policy", manifests.InstallFromFile(policyPath))
	createNetworkProblems.Assess("install dynakube", dynakube.Apply(
		dynakube.NewBuilder().
			WithDefaultObjectMeta().
			FeatureFlag(featureFlag).
			ApiUrl(secretConfigs[0].ApiUrl).
			CloudNative(codeModulesSpec()).
			Build()),
	)
	createNetworkProblems.Assess("install deployment", manifests.InstallFromFile("../testdata/cloudnative/codemodules-deployment.yaml"))
	createNetworkProblems.Assess("start sample apps and injection", sampleapps.Install)
```

File: test/cloudnative/network_problems_test.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+)
+
+func networkProblems(t *testing.T, policyPath string) features.Feature {
+	secretConfigs, err := secrets.DefaultMultiTenant(afero.NewOsFs())
+
+	require.NoError(t, err)
+
+	featureFlag := map[string]string{
+		featureMaxCSIMountAttempts: mountAttempts,
+	}
```
Comment (meik99) [inline test/cloudnative/network_problems_test.go:45]:
```suggestion
	featureFlag := map[string]string{
		featureMaxCSIMountAttempts: mountAttempts,
	}
	secretConfigs, err := secrets.DefaultMultiTenant(afero.NewOsFs())

	require.NoError(t, err)
```

**Review by meik99 — CHANGES_REQUESTED** [review]

File: test/cloudnative/network_problems_test.go [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+)
+
+const (
+	sampleAppDirPath           = "opt/dynatrace/oneagent-paas"
+	errorMessage               = "ERROR: ld.so: object '/opt/dynatrace/oneagent-paas/agent/lib64/liboneagentproc.so' from LD_PRELOAD cannot be preloaded"
+	sleepMinuteAmount          = 5
+	threshold                  = int32(3)
```
Comment (meik99) [inline test/cloudnative/network_problems_test.go:31]:
```suggestion
	agentMountPath           = "opt/dynatrace/oneagent-paas"
	ldPreloadError               = "ERROR: ld.so: object '/opt/dynatrace/oneagent-paas/agent/lib64/liboneagentproc.so' from LD_PRELOAD cannot be preloaded"
	podRestartTimeout          = 5 * time.Minute
	restartCountThreshold                  = int32(3)
```

File: test/cloudnative/network_problems_test.go [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+	featureMaxCSIMountAttempts = "feature.dynatrace.com/max-csi-mount-attempts"
```
Comment (meik99) [inline test/cloudnative/network_problems_test.go:32]: Annotation is already defined as `v1beta1.AnnotationFeatureMaxFailedCsiMountAttempts`

File: test/cloudnative/network_problems_test.go [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+	mountAttempts              = "5"
```
Comment (meik99) [inline test/cloudnative/network_problems_test.go:33]: The default max mount attempt is `10`, I don't think setting this feature flag specifically is really necessary at all

**Review by meik99 — CHANGES_REQUESTED** [review]

File: test/cloudnative/network_problems_test.go (hunk 6 above)
Comment (waodim) [reply test/cloudnative/network_problems_test.go:33]: FYI: when removing the feature flag, which sets the max mount attempt to 10, I have to adapt the timeout in the step where I start the sample apps and start the injection, as 10 mount attempts take about ~8 minutes. I'm completely fine with that, just wanted to let you know.

File: test/cloudnative/network_problems_test.go [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+)
+
+const (
+	agentMountPath           = "opt/dynatrace/oneagent-paas"
```
Comment (chrismuellner) [inline test/cloudnative/network_problems_test.go:28]:
```suggestion
	agentMountPath           = "/opt/dynatrace/oneagent-paas"
```
should be absolute

File: test/cloudnative/network_problems_test.go [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+	createNetworkProblems.Assess("apply network policy", manifests.InstallFromFile(policyPath))
+	createNetworkProblems.Assess("install dynakube", dynakube.Apply(
+		dynakube.NewBuilder().
+			WithDefaultObjectMeta().
+			FeatureFlag(featureFlag).
+			ApiUrl(secretConfigs[0].ApiUrl).
+			CloudNative(codeModulesSpec()).
+			Build()),
+	)
+	createNetworkProblems.Assess("install deployment", manifests.InstallFromFile("../testdata/cloudnative/codemodules-deployment.yaml"))
```
Comment (chrismuellner) [inline test/cloudnative/network_problems_test.go:61]: should this file also be defined in a `const`?

File: test/cloudnative/network_problems_test.go (hunk 7 above)
Comment (waodim) [reply test/cloudnative/network_problems_test.go]: done

File: test/cloudnative/network_problems_test.go (hunk 8 above)
Comment (waodim) [reply test/cloudnative/network_problems_test.go]: done

**Review by meik99 — DISMISSED** [review]

**Review by meik99 — APPROVED** [review]

**Pull request merged**